WordPressComKit/SiteService.swift

```diff
@@ -1,44 +1,82 @@
 import Foundation
 import Alamofire
 
-public class SiteService {
-    public init() {}
-    
-    public func fetchSite(siteID: Int, completion: (Site?, NSError?) -> Void) {
-        Alamofire
-            .request(RequestRouter.Site(siteID: siteID))
-            .validate()
-            .responseJSON { response in
-                guard response.result.isSuccess else {
-                    completion(nil, response.result.error)
-                    return
-                }
-                
-                let json = response.result.value as? [String: AnyObject]
-                let site = self.mapJSONToSite(json!)
-                
-                completion(site, nil)
+extension Result {
+    func map<T>(transform: Value -> T) -> Result<T, Error> {
+        return flatMap({ .Success(transform($0)) })
+    }
+
+    func flatMap<T>(transform: Value -> Result<T, Error>) -> Result<T, Error> {
+        switch self {
+        case .Success(let value):
+            return transform(value)
+        case .Failure(let error):
+            return .Failure(error)
         }
     }
-    
-    public func fetchSites(showActiveOnly: Bool = true, completion:([Site]?, NSError?) -> Void) {
+}
+
+// TODO: use a decent error
+let jsonDecodeError = NSError(domain: "WordPressComKit", code: -1, userInfo: nil)
+
+public typealias JSON = [String: AnyObject]
+
+public protocol Requester {
+    func requestJSON(request: NSURLRequest, completion: Result<JSON, NSError> -> Void)
+}
+
+struct AlamofireRequester: Requester {
+    func requestJSON(request: NSURLRequest, completion: Result<JSON, NSError> -> Void) {
         Alamofire
-            .request(RequestRouter.Sites(showActiveOnly: showActiveOnly))
+            .request(request)
             .validate()
```

Comment:
This does obscure the ability to handle validation differently between endpoints. An example of validate with a parameter:

```swift
         .validate(statusCode: 200..<300)
         .validate(contentType: ["application/json"])
```